File: app/src/main/java/com/badlogic/masaki/firebasesample/di/AppComponent.java
package com.badlogic.masaki.firebasesample.di;

import android.app.Application;
import com.badlogic.masaki.firebasesample.App;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivityBindingModule.class,
        AppModule.class,
        ExecutorModule.class,
        DatabaseModule.class
        })
public interface AppComponent extends AndroidInjector<App> {
    @Component.Builder
    interface Builder {
        @BindsInstance Builder application(Application application);
        AppComponent build();
    }
}

File: app/src/main/java/com/badlogic/masaki/firebasesample/di/AppComponent.kt
package com.badlogic.masaki.firebasesample.di

import android.app.Application
import com.badlogic.masaki.firebasesample.App
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ActivityBindingModule::class,
    AppModule::class,
    ExecutorModule::class,
    DatabaseModule::class
])
interface AppComponent : AndroidInjector<App> {
    @Component.Builder
    interface Builder {
        @BindsInstance fun application(application: Application): Builder
        fun build(): AppComponent
    }
}